src/frontend/apps/desk/src/features/teams/member-management/components/MemberGrid.tsx

```diff
@@ -43,30 +48,64 @@ const defaultOrderingMapping: Record<string, string> = {
  */
 function formatSortModel(
   sortModel: SortModelItem,
-  mapping = defaultOrderingMapping,
-) {
+  mapping: Record<string, string> = defaultOrderingMapping,
+): string {
   const { field, sort } = sortModel;
   const orderingField = mapping[field] || field;
   return sort === 'desc' ? `-${orderingField}` : orderingField;
 }
 
+const MS_DEBOUNCE = 500;
+export function useDebounce(value: string, delay: number = MS_DEBOUNCE) {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+
+  return debouncedValue;
+}
+
```

Comment:
Quite usefull, it could have its own file (`src/api/hooks` or `src/hooks`), with a small jest test ?